Neos.ContentRepository.Core/Classes/EventStore/DecoratedEvent.php

```diff
@@ -17,7 +19,9 @@ final class DecoratedEvent
     private function __construct(
         public readonly EventInterface $innerEvent,
         public readonly EventId $eventId,
-        public readonly EventMetadata $eventMetadata,
+        public readonly ?EventMetadata $eventMetadata = null,
+        public readonly ?CausationId $causationId = null,
+        public readonly ?CorrelationId $correlationId = null,
```

Comment:
Is the introduction of these two properties releated to this issue? https://github.com/neos/neos-development-collection/issues/3887

Reply:
Yes, slightly. Previously causation and correlation id were (implicitly) part of the event metadata. With https://github.com/neos/eventstore/commit/4dacb9abbe75b06f76c06dddc3a045416e7e8b7d they are now a core concept so we have to reflect that.

But it does not resolve the issue in that we don't use causation and correlation ids consistently yet in the ESCR